spec/engine_spec.rb

```diff
@@ -0,0 +1,28 @@
+require_relative '../lib/engine'
+require_relative '../lib/game'
+require_relative '../lib/player'
+
+describe Engine do
+  let(:engine) { Engine.new }
+  let(:firstPlayer) { Player.new('X', 'John') }
+  let(:secondPlayer) { Player.new('O', 'John') }
+
+  describe '#winning' do
+    it 'returns true when there is a winner' do
+      board = [
+        firstPlayer, firstPlayer, firstPlayer,
+        secondPlayer, secondPlayer, nil,
+        nil, nil, nil
+      ]
+
+      expect(engine.winning?(board, 3)).to eql(true)
+    end
+
+    it 'returns false when there is no winner' do
+      board = Array.new(9)
+      board[0] = firstPlayer
+
+      expect(engine.winning?(board, 1)).to eql(false)
+    end
+  end
```

Comment:
This method is calling some private methods you have on the Engine class.  Although there is no need to test private methods, you do need to test your winning method for all that it is checking. So you need to:
- check for a winner in a row
- check for a winner in a column
- check for a winner in a diagonal

Reply:
Done